File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/C++/s826491904_commented.cpp
#include <iostream>
#include <algorithm>

using namespace std;

class Main {
public:
    void solve() {
        int N;
        cin >> N;
        int max, min;
        cin >> max;
        min = max;
        for (int i = 1; i < N; i++) {
            int in;
            cin >> in;
            max = std::max(max, in);
            min = std::min(min, in);
        }
        cout << max - min;
    }
};

int main() {
    Main().solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/inline/uncomment/direct/Java/Go/s826491904_commented.go
package main

import (
	"fmt"
)

func main() {
	var N int
	fmt.Scan(&N)
	var max, min int
	fmt.Scan(&min)
	max = min
	for i := 1; i < N; i++ {
		var in int
		fmt.Scan(&in)
		if in > max {
			max = in
		}
		if in < min {
			min = in
		}
	}
	fmt.Println(max - min)
}
// End of Code
